torch/autograd/_functions/pointwise.py

```diff
@@ -61,22 +62,25 @@ def backward(self, grad_output):
 
 class Sigmoid(InplaceFunction):
 
-    def forward(self, i):
-        if self.inplace:
-            self.mark_dirty(i)
+    @staticmethod
+    def forward(ctx, i, inplace=False):
+        ctx.input_size = i.size()
+        if inplace:
+            ctx.mark_dirty(i)
             result = i.sigmoid_()
         else:
             result = i.sigmoid()
-        self.save_for_backward(result)
+        ctx.save_for_backward(result)
         return result
 
-    def backward(self, grad_output):
-        result, = self.saved_tensors
-        grad_input = grad_output.new()
-        backend = type2backend[type(result)]
-        backend.Sigmoid_updateGradInput(backend.library_state, None, grad_output,
-                                        grad_input, result)
-        return grad_input
+    @staticmethod
+    def backward(ctx, grad_output):
+        result, = ctx.saved_variables
+        grad_input = Variable(grad_output.data.new(ctx.input_size).zero_())
+        backend = type2backend[type(result.data)]
+        backend.Sigmoid_updateGradInput(backend.library_state, None, grad_output.data,
```

Comment:
Re-packing the grad in a new Variable doesn't retain the saved history so it's wrong. When a grad is volatile, then it doesn't have any history anyway, so this faster method is valid in this case.